source/extensions/filters/http/wasm/wasm_filter.h

```diff
@@ -31,17 +31,19 @@ class FilterConfig : Logger::Loggable<Logger::Id::wasm> {
     if (handle.has_value()) {
       wasm = handle->wasm().get();
     }
-    if (plugin_->fail_open_ && (!wasm || wasm->isFailed())) {
-      return nullptr;
+    if (!wasm || wasm->isFailed()) {
```

Comment:
question: is this change something to do with https://github.com/envoyproxy/envoy/issues/14947?

Reply:
No, there are functional changes here, only style.

Previously, we had:
```
if (plugin_->fail_open_ && (!wasm || wasm->isFailed())) {
  return nullptr;
}
```
which handled the fail open case, and
```
return std::make_shared<Context>(wasm, root_context_id_, plugin_);
```
which handled both the valid case, and the fail closed case (where `wasm == nullptr` and the rest didn't matter).

I split the latter to make it more readable, since the new version is:
```
return std::make_shared<Context>(wasm, handle->configId(), plugin_);
```
and `handle` isn't valid in case of fail closed case.

Alternatively, I could undo the style changes and turn this into:
```
return std::make_shared<Context>(wasm, handle.has_value() ? handle->configId() : 0, plugin_);
```
but that seems more confusing than having separate codepath.

Let me know what you prefer.